### ISSUE TITLE
Job execution is not visible anymore in job list after execution 

### ISSUE THREAD

**Issue by pierreberton** [hashicorp/terraform-provider-kubernetes#941]
### Terraform Version and Provider Version

Terraform : v0.12.28
Provider Kubernetes : v1.11.3 (OK) - v1.11.4+ (KO)
Cloud : Orange Business Service (Flexible engine)

### Affected Resource(s)

- kubernetes_job

### Terraform Configuration Files
```hcl
resource "kubernetes_job" "job-init-flyway" {
  metadata {
    name = "job-init-flyway"
  }
  spec {
    template {
      metadata {}
      spec {
        container {
          name  = "init-flyway"
          image = "flyway/flyway:6.5.0"
          args = [
            "-url=jdbc:postgresql://my-service:5432/mydb",
            "-user=user",
            "-password=${random_password.db-password.result}",
            "-baselineVersion=-1",
            "baseline"
          ]
        }
        restart_policy = "Never"
      }
    }
    backoff_limit = 1
  }
}
```
NOTE : it can be reproduced with any container, mine (flyway) is just for example

### Debug Output

All is good

### Panic Output

All is good

### Expected Behavior
What should have happened?
After job execution, job is still visible in job list

### Actual Behavior
What actually happened?
With Kubernetes provider v1.11.3, jobs stay visible after execution in job list
With Kubernetes provider v1.11.4 or newer, jobs are not visible anymore after execution in job list and it causes jobs to be executed each time we launch `terraform apply`

### Steps to Reproduce

1. `terraform apply --auto-approve`

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by almo-ltoeroek** [comment]
I am having the exact same issue on K8s 1.17, strangely not an issue on K8s 1.11

**Comment by aareet** [comment]
This should have been fixed by #1053, please let us know if that is not the case.
